ddtrace/llmobs/experimentation/_experiment.py

```diff
@@ -0,0 +1,1242 @@
+                            "duration": time.time() - start_time,
+                            "dataset_record_index": idx,
+                            "project_name": self.project_name,
+                            "experiment_name": self.name,
+                            "dataset_name": self.dataset.name,
+                            "span_id": span_id,
+                            "trace_id": trace_id,
+                        },
+                        "error": {"message": None, "stack": None, "type": None},
+                    }
+                except Exception as e:
+                    error_message = str(e)
+                    span.error = 1
+                    span.set_exc_info(type(e), e, e.__traceback__)
+
+                    LLMObs.annotate(
+                        span,
+                        input_data=input_data,
+                        tags={
+                            "dataset_id": self.dataset._datadog_dataset_id,
+                            "dataset_record_id": row.get("record_id"),
+                            "experiment_id": self._datadog_experiment_id,
+                        },
+                    )
+                    LLMObs._tag_expected_output(span, expected_output)
+
+                    if idx % FLUSH_EVERY == 0:
+                        LLMObs.flush()
+
+                    return {
+                        "idx": idx,
+                        "output": None,
+                        "metadata": {
+                            "timestamp": start_time,
+                            "duration": time.time() - start_time,
+                            "dataset_record_index": idx,
+                            "project_name": self.project_name,
+                            "experiment_name": self.name,
+                            "dataset_name": self.dataset.name,
+                            "span_id": span_id,
+                            "trace_id": trace_id,
+                        },
+                        "error": {
+                            "message": error_message,
+                            "stack": traceback.format_exc(),
+                            "type": type(e).__name__,
+                        },
+                    }
+
+        _jobs = DEFAULT_CONCURRENT_JOBS if (sample_size and jobs == DEFAULT_CONCURRENT_JOBS) else jobs
+        with concurrent.futures.ThreadPoolExecutor(max_workers=_jobs) as executor:
+            for result in executor.map(process_row, enumerate(subset_dataset)):
+                outputs_buffer.append(result)
+                progress.update(error=result["error"]["message"] is not None)
+                if raise_errors and result["error"]["message"]:
+                    raise RuntimeError(
+                        f"Error on record {result['idx']}: {result['error']['message']}\n"
+                        f"Stack Trace:\n{result['error']['stack']}"
+                    )
+
+        LLMObs.flush()
+        time.sleep(API_PROCESSING_TIME_SLEEP)
```

Comment:
Why sleep and flush here?

Reply:
Flush is to ensure spans get submitted if in serverless env (task ends). Fine with force flushing for now, but sleeping is weird

Reply:
Should replace sleep with more verbose message `(your trace/experiment will be available momentarily at ...)`

Reply:
weird question but here is just for submitting spans no? Why would we need to print a link to the experiment here? Shouldn't we be printing the link once the evals are finished/submitted?

Reply:
I randomly stumbled here, but if y'all identify weird bits of logic to accommodate back-end behaviors, let me know and we can improve it. I agree that putting a sleep anywhere is very weird and if it's just to generate a link, we should remove it. These small nice to haves can come later.